dev-embedded/jal/files/jal-0.4.62-fix-incompatible-integer-to-pointer-clang16.patch

```diff
@@ -0,0 +1,24 @@
+Bug: https://bugs.gentoo.org/883243
+--- a/jal/treerep.c
++++ b/jal/treerep.c
+@@ -892,7 +892,7 @@ tree new_asm(loc_t loc, opcode_t opcode, tree arg1, int arg2)
+ 
+ 
+     if (code_has(p->opcode, field_flabel)) {
+-        assert_pointer(NULL, arg2);
++        assert_pointer(NULL, &arg2);
+         p->next = new_const(new_value(type_universal, arg2));
+     }
+ 
+--- a/jal/treetools.c
++++ b/jal/treetools.c
+@@ -600,7 +600,8 @@ tree arg(tree p, int n, boolean must_find)
+ #endif
+     assert_kind(p->loc, p, node_procedure);
+ /*    jal_assert(p->loc, n > 0); */
+-    assert_pointer(p->loc, n > 0);
++	int temp_val = (n > 1);
++    assert_pointer(p->loc, &temp_val);
```

Comment:
This is also wrong, but by the looks of it the original is also wrong.  I don't know why they commented out `jal_assert`, it looks like it should have been correct in the first place.  Probably it didn't work?